src/ai/assistant.ts

```diff
@@ -0,0 +1,440 @@
+    const result = await sendTx(this.ain, txBody);
+
+    return { ...result, assistant: data };
+  }
+
+  /**
+   * Deletes an assistant.
+   * @param {string} objectId - The ID of AINFT object.
+   * @param {string} tokenId - The ID of AINFT token.
+   * @param {string} assistantId - The ID of assistant.
+   * @returns Returns a promise that resolves with both the transaction result and the deleted assistant.
+   */
+  async delete(
+    objectId: string,
+    tokenId: string,
+    assistantId: string
+  ): Promise<AssistantDeleteTransactionResult> {
+    const address = this.ain.signer.getAddress();
+
+    await validateObject(this.ain, objectId);
+    const isAdmin = await isObjectOwner(this.ain, objectId, address);
+    await validateToken(this.ain, objectId, tokenId);
+    await validateAssistant(this.ain, objectId, tokenId, assistantId);
+
+    const serverName = this.ainize.getServerName();
+    await validateServerConfigurationForObject(this.ain, objectId, serverName);
+
+    const opType = OperationType.DELETE_ASSISTANT;
+    const body = { assistantId };
+    const { data } = await this.ainize.requestWithAuth<AssistantDeleted>(this.ain, {
+      serverName,
+      opType,
+      data: body,
+    });
+
+    if (!isAdmin) {
+      return { delAssistant: data };
+    }
+
+    const txBody = this.buildTxBodyForDeleteAssistant(objectId, tokenId, address);
+    const result = await sendTx(this.ain, txBody);
+
+    return { ...result, delAssistant: data };
+  }
+
+  /**
+   * Retrieves an assistant.
+   * @param {string} objectId - The ID of AINFT object.
+   * @param {string} tokenId - The ID of AINFT token.
+   * @param {string} assistantId - The ID of assistant.
+   * @returns Returns a promise that resolves with the assistant.
+   */
+  async get(objectId: string, tokenId: string, assistantId: string): Promise<Assistant> {
+    await validateObject(this.ain, objectId);
+    await validateToken(this.ain, objectId, tokenId);
+    await validateAssistant(this.ain, objectId, tokenId, assistantId);
+
+    const serverName = this.ainize.getServerName();
+    await validateServerConfigurationForObject(this.ain, objectId, serverName);
+
+    const opType = OperationType.RETRIEVE_ASSISTANT;
+    const body = { assistantId };
+    const { data } = await this.ainize.requestWithAuth<Assistant>(this.ain, {
+      serverName,
+      opType,
+      data: body,
+    });
+
+    return data;
+  }
+
+  async list(objectId: string, { limit = 20, offset = 0, order = 'desc' }: QueryParams) {
+    const address = this.ain.signer.getAddress();
+
+    await validateObject(this.ain, objectId);
+
+    const tokens = await this.getTokensByAddress(objectId, address);
+    const assistants = this.getAssistantsFromTokens(tokens);
+    const sorted = this.sortAssistants(assistants, order);
+
+    const total = sorted.length;
+    const items = sorted.slice(offset, offset + limit);
+
+    return { total, items };
+    // NOTE(jiyoung): example data
+    /**
```

Comment:
minor) how about just using '/*' as it's not a tsdoc comment?

Reply:
thank you! i made mistake with this syntax